test/com/amazon/ionelement/IonOperatorsTests.kt

```diff
@@ -0,0 +1,386 @@
+package com.amazon.ionelement
+
+import com.amazon.ion.Decimal
+import com.amazon.ionelement.api.*
+import java.math.BigInteger
+import kotlin.random.Random
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import org.junit.jupiter.api.Test
+
+@OptIn(IonOperators::class)
+class IonOperatorsTests {
```

Comment:
None of your tests appear to have any annotations on the values. Can you add test cases for when annotations are expected to be preserved?

Additionally, having many small tests is preferable to having few giant tests because it's easier to pinpoint a failure when something goes wrong. Could you try to split them up into more discrete units?

Reply:
Its not obvious but at the bottom of the test file all the `ion*()` methods that create the Ion values are overloaded to add annotations and metas:

```
private val annotations = listOf("annotation1", "annotation2")
private val metas = mapOf("key1" to "value1", "key2" to listOf("value2", "value3"), "key3" to 1)

private fun ionBool(x: Boolean) = ionBool(x, annotations, metas)
private fun ionInt(x: Long) = ionInt(x, annotations, metas)
private fun ionInt(x: BigInteger) = ionInt(x, annotations, metas)
private fun ionFloat(x: Double) = ionFloat(x, annotations, metas)
private fun ionDecimal(x: Decimal) = ionDecimal(x, annotations, metas)
private fun ionString(x: String) = ionString(x, annotations, metas)
private fun ionSymbol(x: String) = ionSymbol(x, annotations, metas)
```

Reply:
I will work on splitting up the tests